satellite/gracefulexit/endpoint.go

```diff
@@ -549,11 +551,17 @@ func (endpoint *Endpoint) processIncomplete(ctx context.Context, stream processS
 
 	pieceSize := eestream.CalcPieceSize(pointer.GetSegmentSize(), redundancy)
 
+	excludedIPs, err := endpoint.overlay.GetNodeIPs(ctx, excludedNodeIDs)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
 	request := overlay.FindStorageNodesRequest{
 		RequestedCount: 1,
 		FreeBandwidth:  pieceSize,
 		FreeDisk:       pieceSize,
 		ExcludedNodes:  excludedNodeIDs,
+		ExcludedIPs:    excludedIPs,
```

Comment:
Shouldn' t find storage nodes already handle this based on the excluded nodes?

Reply:
Feels weird to do multiple queries to the database to ensure this.

Reply:
No, FindStorageNodes doesn't look up the IPs of the excluded nodes and further exclude nodes that match those same IPs. That would be nice, but might be fairly complicated to do efficiently. We'd risk slowing down node selection in all cases for the sake of this specialized case that's only needed in gracefulexit (ok, maybe also in repair).

Reply:
Can we have a wrapper for the FindStorageNodes to do that and only repair and gracefulexit would use it?

Reply:
I think that would make sense, and now that I think about it, maybe that's what Egon was suggesting here: https://github.com/storj/storj/pull/3435#discussion_r341367578

Reply:
Initially was thinking of making FindStorageNodesRequest to handle this case, but I think making it a method on overlay.Service might be easier and a neater solution. We can later optimize, if we need to.